libs/blockchain/src/blockchain-rpc-query.service.ts

```diff
@@ -237,6 +245,23 @@ export class BlockchainRpcQueryService extends PolkadotApiService {
     };
   }
 
+  public async checkTxCapacityLimit(providerId: AnyNumber, encodedExt: HexString): Promise<boolean> {
+    const capacityInfo = await this.capacityInfo(providerId);
+    const { remainingCapacity } = capacityInfo;
+
+    // Calculate the total capacity cost for all encoded extensions
+    let totalAdjustedWeightFee = BigInt(0);
+    const capacityCost = await this.getCapacityCostForExt(encodedExt);
+    if (capacityCost.inclusionFee.isSome) {
+      const inclusionFee = capacityCost.inclusionFee.unwrap();
+      totalAdjustedWeightFee += inclusionFee.adjustedWeightFee.toBigInt();
+    }
```

Comment:
Should maybe put this in a try/catch so a call for an invalid extrinsic can return false instead of throw an error

Reply:
Good point, thought the rpc handled wrong input and return free detail will none but will put a try catch